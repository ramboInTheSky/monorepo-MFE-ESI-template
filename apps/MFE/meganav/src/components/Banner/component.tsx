import styled from "styled-components"

import {breakpoints} from "@monorepo/themes"

export const Container = styled.div`
    height: 4rem;
    text-align: center;
    height: 6.25rem;
    @media (min-width: ${breakpoints.values.lg}px) {
        height: 2.5rem;
    }
    a,
    img {
        width: 100%;
        height: 100%;
    }
`
