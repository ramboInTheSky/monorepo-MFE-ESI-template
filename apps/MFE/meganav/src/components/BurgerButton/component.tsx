import styled from "styled-components"

export const Button = styled.button`
    background: none;
    color: inherit;
    border: none;
    padding: 0;
    font: inherit;
    cursor: pointer;
    outline: inherit;
    order: 1;
    width: 2.75rem;
    height: 2.75rem;
    align-items: center;
    display: flex;
    justify-content: center;
`
