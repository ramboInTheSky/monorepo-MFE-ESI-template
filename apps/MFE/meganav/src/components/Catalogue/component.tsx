import styled from "styled-components"
import {breakpoints} from "@monorepo/themes"

export const Hr = styled.hr`
    background: #eaeaea;
    margin: 0;
    padding: 0;
    border: none;
    height: 1px;
    @media (min-width: ${breakpoints.values.lg}px) {
        display: none;
    }
`
export const Container = styled.div<{hasMissions: boolean}>`
    width: 100%;
    overflow-x: hidden;
    overflow-y: hidden;
    padding-bottom: ${({hasMissions}) => (hasMissions ? 0 : "4rem")};
    @media (min-width: ${breakpoints.values.lg}px) {
        overflow-y: auto;
        max-height: calc(100vh - 6rem);
        padding: 1rem;
        padding-bottom: 0;
        display: flex;
        flex-direction: row;
        justify-content: space-between;
    }
`

type ColumnProps = {
    numberofcolumns: number
}

export const Column = styled.div<ColumnProps>`
    display: flex;
    flex-direction: column;
    @media (min-width: ${breakpoints.values.lg}px) {
        width: ${props => `calc(100% / ${props.numberofcolumns})`};
        height: 100%;
        padding-right: 1rem;
    }
`
