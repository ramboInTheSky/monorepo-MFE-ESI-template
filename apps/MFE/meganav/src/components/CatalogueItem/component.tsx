import styled from "styled-components"
import {breakpoints} from "@monorepo/themes"

type AnchorProps = {
    linkColour: string | null
    fontFamily: string | null
    fontWeight: string
    isDirectLink: boolean
}

export const Container = styled.li<AnchorProps>`
    display: flex;
    list-style: none;
    margin-bottom: 1rem;
    width: 100%;

    a {
        display: flex;
        justify-content: space-between;
        text-decoration: none;
        font-size: ${({isDirectLink}) => (isDirectLink ? "0.9375rem" : "0.875rem")};
        line-height: 1.5;
        letter-spacing: 0.12px;
        word-break: break-word;
        color: ${({theme, linkColour}) => linkColour || theme.colours.text.default};
        font-family: ${({theme, fontFamily}) =>
            fontFamily || `${theme.colours.font.secondary.regular.family}, ${theme.colours.font.default}`};
        font-weight: ${({fontWeight}) => fontWeight || "500"};
        @media (min-width: ${breakpoints.values.lg}px) {
            font-size: 0.8125rem;
        }
    }

    a:hover {
        text-decoration: underline;
    }

    @media (min-width: ${breakpoints.values.sm}px) {
        width: ${({isDirectLink}) => (isDirectLink ? "100%" : "50%")};
    }

    @media (min-width: ${breakpoints.values.lg}px) {
        width: 100%;
        line-height: 1.5;
        margin-bottom: 0.5rem;
    }
`

export const Image = styled.img`
    margin: 0 1rem 0 0.25rem;
    width: 1.875rem;
    height: 0.9375rem;
`

export const TitleAndImageContainer = styled.div`
    display: flex;
    align-items: center;
`
