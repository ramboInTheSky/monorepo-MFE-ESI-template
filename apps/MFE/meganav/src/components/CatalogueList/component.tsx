import styled from "styled-components"
import {breakpoints} from "@monorepo/themes"
import {Hidden as MUIHidden, AccordionSummary, Accordion, AccordionDetails} from "@mui/material"
import {DEFAULT_IMAGE_PLACEHOLDER} from "../../config/constants"

type TitleProps = {
    linkColour: string
    fontFamily: string | null
    fontWeight: string
    isFirstElement?: boolean
}

export const AccordionPanel = styled(Accordion)`
    background: none;
    box-shadow: none;
    width: 100%;
    border-bottom: ${props => props.theme.colours.utilities.divider};
`
export const AccordionPanelDetails = styled(AccordionDetails)`
    padding: 0;
`
// FIXME replace AccordionPanelSummary background: ${props => props.theme.colours.utilities.backgroundAccent};
export const AccordionPanelSummary = styled(AccordionSummary)`
    && {
        background: ${props => props.theme.colours.utilities.backgroundAccent};
        @media (min-width: ${breakpoints.values.md}px) {
            background: #ffffff;
        }
        padding: 0;
        min-height: auto;
        width: 100%;
        && > div {
            margin: 0;
        }
        > div:nth-child(2) {
            padding: 1rem;
        }

        img {
            width: 0.75rem;
        }
    }
`
export const Hidden = styled(MUIHidden)`
    width: 100%;
`

export const Container = styled.div`
    display: flex;
    flex-direction: column;
    width: 100%;
    align-items: start;
    @media (min-width: ${breakpoints.values.lg}px) {
        border-top: 0;
    }
`
export const Title = styled.div<TitleProps>`
    height: 2.75rem;
    padding-left: 1rem;
    width: 100%;
    display: flex;
    flex-direction: row;
    align-items: center;
    margin: 0;
    font-size: 0.9375rem;
    letter-spacing: 0.12px;
    word-break: break-word;
    color: ${({theme, linkColour}) => linkColour || theme.colours.text.default};
    font-family: ${({theme, fontFamily}) =>
        fontFamily || `${theme.colours.font.primary.regular.family}, ${theme.colours.font.default}`};
    font-weight: ${({fontWeight}) => fontWeight || "500"};

    @media (min-width: ${breakpoints.values.md}px) {
        padding-left: 2rem;
    }
    @media (min-width: ${breakpoints.values.lg}px) {
        height: unset;
        padding-left: 0;
        font-size: 0.875rem;
        margin: ${({isFirstElement}) => (isFirstElement ? "0 0 0.75rem 0" : "1.25rem 0 0.75rem")};
        background: none;
    }
`
export const List = styled.ul`
    display: flex;
    flex-direction: row;
    margin: 0;
    padding: 0;
    align-items: start;
    padding: 0.813rem 1rem 0 1rem;
    height: inherit;
    flex-wrap: wrap;
    width: 100%;
    @media (min-width: ${breakpoints.values.md}px) {
        padding-left: 2rem;
    }
    @media (min-width: ${breakpoints.values.lg}px) {
        padding: 0;
        flex-direction: column;
        flex-wrap: nowrap;
    }
`

export const ImagePlaceholder = styled.span<{url?: string}>`
    && {
        width: 2rem;
        height: 2rem;
        border-radius: 50%;
        background-image: url(${({url}) => url || DEFAULT_IMAGE_PLACEHOLDER}), url(${DEFAULT_IMAGE_PLACEHOLDER});
        background-size: cover;
        background-repeat: no-repeat;
        margin: 0 0.75rem 0 0;
        flex-shrink: 0;
    }
`
