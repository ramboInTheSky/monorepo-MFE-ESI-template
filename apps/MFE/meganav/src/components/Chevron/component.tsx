import styled from "styled-components"
import {breakpoints} from "@monorepo/themes"

export type GradientProps = {
    flip: boolean
    isShown: boolean
}

const gradientColor = (background: string): string =>
    `linear-gradient(to right, ${background}, rgba(30, 30, 30, 0) 100%)`

export const Gradient = styled.div<GradientProps>`
    transition: width 0.15s;
    z-index: 3;
    width: ${({isShown}) => (isShown ? "2rem" : "0")};
    height: 100%;
    transform: rotateZ(${({flip}) => (flip ? "180deg" : "0")});
    margin-left: ${({flip}) => (flip ? "-1rem" : "0")};
    margin-right: ${({flip}) => (flip ? "0" : "-1rem")};
    background: ${props => gradientColor(props.theme.colours.header.navLowerBackground.default)};
    @media (min-width: ${breakpoints.values.lg}px) {
        background: ${props => gradientColor(props.theme.colours.header.navLowerBackground.desktop)};
    }
`
export type ContainerProps = {doRotate: boolean}
export const Container = styled.div<ContainerProps>`
    pointer-events: none;
    width: 1rem;
    height: 100%;
    display: flex;
    justify-content: left;
    align-items: center;
    transform: rotateY(${({doRotate}) => (doRotate ? "180deg" : "0")});
    z-index: 3;
    && button {
        pointer-events: auto;
        width: 1.5rem;
        border-radius: 0;
        height: 100%;
        padding: 0;
        span:first-child {
            width: auto;
        }
        background: ${props => props.theme.colours.header.navLowerBackground.default};
        @media (min-width: ${breakpoints.values.lg}px) {
            background: ${props => props.theme.colours.header.navLowerBackground.desktop};
        }
    }
`
