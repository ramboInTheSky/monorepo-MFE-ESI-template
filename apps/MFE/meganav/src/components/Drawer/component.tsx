import styled from "styled-components"
import {Drawer as MUIDrawer} from "@mui/material"

/*
 * Title: Using to style top !important
 * problem: we do not want to use !important but we want a modal and a modal is expected to cover the entire screen
 * in this case however, material UI probably assumes you'll have the entire screen covered and uses inline styles
 * to define this behaviour. But here the modal as it is the case of next, partially exposes the header area hence it
 * does not fully cover the entire screen.
 * Solution: for now we have to use !important
 * Refs:
 *    https://github.com/mui-org/material-ui/issues/16441
 *    https://github.com/mui-org/material-ui/issues/16442
 */

export const DrawerContainer = styled(MUIDrawer)`
    && {
        position: relative !important;
        height: 100vh;
    }

    &&.drawer > div:nth-child(1) {
        position: absolute;
        background: rgba(0, 0, 0, 0.9);
        z-index: 0;
    }
    &&.drawer > div:nth-child(3) {
        box-shadow: none;
        position: absolute;
        width: calc(100% - 3rem);
    }
`

export const CloseIcon = styled.img`
    position: fixed;
    top: 2.75rem;
    left: 0;
    cursor: pointer;
    height: 2.75rem;
    width: 2.75rem;
`
