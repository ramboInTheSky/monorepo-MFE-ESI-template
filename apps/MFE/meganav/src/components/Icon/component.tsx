import styled from "styled-components"

export type ContainerProps = {
    rotationDegrees: number
    width: number
}

export const Container = styled.img<ContainerProps>`
    transform: ${({rotationDegrees}) => `rotateZ(${rotationDegrees}deg)`};
    cursor: pointer;
    width: ${({width}) => `${width}rem`};
`
