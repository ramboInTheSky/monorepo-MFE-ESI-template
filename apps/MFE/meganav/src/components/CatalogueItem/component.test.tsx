import React from "react"
import {SCThemeProvider} from "@monorepo/theme-provider"
import {render} from "@testing-library/react"
import {Container} from "./component"
import {mockTheme} from "../../../__mocks__/mockStore"

describe("CatalogueItem component: ", () => {
    describe("Li ", () => {
        const props = {
            linkColour: "#fff",
            fontFamily: "",
            fontWeight: "",
            isDirectLink: true,
        }
        it("should match the snapshot ", () => {
            const {asFragment} = render(
                <SCThemeProvider theme={mockTheme}>
                    <Container {...props} />
                </SCThemeProvider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })
        it("should match the snapshot if isDirectLink is false", () => {
            props.isDirectLink = false
            const {asFragment} = render(
                <SCThemeProvider theme={mockTheme}>
                    <Container {...props} />
                </SCThemeProvider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })
        it("should match the snapshot if there is no linkColour", () => {
            const {asFragment} = render(
                <SCThemeProvider theme={mockTheme}>
                    <Container {...props} linkColour={null} />
                </SCThemeProvider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })
    })
})
