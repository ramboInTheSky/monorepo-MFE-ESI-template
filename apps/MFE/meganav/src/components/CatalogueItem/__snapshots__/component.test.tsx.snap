// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogueItem component:  Li  should match the snapshot  1`] = `
<DocumentFragment>
  <li
    class="component__Container-sc-1cuh8iw-0 iyQmcU"
    font-family=""
    font-weight=""
  />
</DocumentFragment>
`;

exports[`CatalogueItem component:  Li  should match the snapshot if isDirectLink is false 1`] = `
<DocumentFragment>
  <li
    class="component__Container-sc-1cuh8iw-0 fCDzjm"
    font-family=""
    font-weight=""
  />
</DocumentFragment>
`;

exports[`CatalogueItem component:  Li  should match the snapshot if there is no linkColour 1`] = `
<DocumentFragment>
  <li
    class="component__Container-sc-1cuh8iw-0 fpVIaw"
    font-family=""
    font-weight=""
  />
</DocumentFragment>
`;
